src/gui/symbology-ng/qgsgraduatedsymbolrendererv2widget.cpp

```diff
@@ -516,15 +538,33 @@ void QgsGraduatedSymbolRendererV2Widget::updateUiFromRenderer( bool updateCount
     updateGraduatedSymbolIcon();
   }
 
+  if ( mGraduatedSymbol )
+  {
+    mSizeUnitWidget->blockSignals( true );
+    mSizeUnitWidget->setUnit( mGraduatedSymbol->outputUnit() );
+    mSizeUnitWidget->setMapUnitScale( mGraduatedSymbol->mapUnitScale() );
+    mSizeUnitWidget->blockSignals( false );
+  }
+
   // set source color ramp
   if ( mRenderer->sourceColorRamp() )
   {
+    tabWidget->setCurrentIndex( 0 );
     cboGraduatedColorRamp->setSourceColorRamp( mRenderer->sourceColorRamp() );
     cbxInvertedColorRamp->setChecked( mRenderer->invertedColorRamp() );
   }
+  else
```

Comment:
rather than rely on sourceColorRamp() to test if the graduated renderer is modifying colour or size, I think a proper QgsGraduatedSymbolRendererV2::graduatedMethod() and setGraduatedMethod should be added (along with an enum of graduated methods, Color and SymbolSize ). This would be more robust if potentially other methods were added in future.